File: app/src/main/java/com/cvgenerator/ui/screens/SkillsScreen.kt
package com.cvgenerator.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cvgenerator.data.*
import com.cvgenerator.ui.theme.Gold
import com.cvgenerator.ui.theme.MediumGray

@Composable
fun SkillsScreen(viewModel: CvViewModel) {
    val cvData by viewModel.cvData.collectAsState()
    var skillName by remember { mutableStateOf("") }
    var skillLevel by remember { mutableStateOf(SkillLevel.INTERMEDIATE) }
    var langName by remember { mutableStateOf("") }
    var langLevel by remember { mutableStateOf("") }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 20.dp, vertical = 16.dp),
        verticalArrangement = Arrangement.spacedBy(20.dp)
    ) {
        Text("Compétences & Langues", fontSize = 20.sp, fontWeight = FontWeight.Bold, color = Gold)

        // Skills section
        Text("Compétences", fontWeight = FontWeight.SemiBold, fontSize = 16.sp)

        cvData.skills.forEach { skill ->
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column(modifier = Modifier.weight(1f)) {
                    Row(
                        horizontalArrangement = Arrangement.SpaceBetween,
                        modifier = Modifier.fillMaxWidth().padding(end = 8.dp)
                    ) {
                        Text(skill.name, fontSize = 14.sp)
                        Text(skill.level.label, fontSize = 12.sp, color = Gold)
                    }
                    Spacer(Modifier.height(4.dp))
                    LinearProgressIndicator(
                        progress = { skill.level.value },
                        modifier = Modifier.fillMaxWidth().height(4.dp),
                        color = Gold,
                        trackColor = MediumGray
                    )
                }
                IconButton(onClick = { viewModel.removeSkill(skill.id) }) {
                    Icon(Icons.Default.Close, contentDescription = null, tint = MaterialTheme.colorScheme.error, modifier = Modifier.size(16.dp))
                }
            }
        }

        // Add skill form
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(MediumGray, RoundedCornerShape(12.dp))
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Text("Ajouter une compétence", fontSize = 14.sp, fontWeight = FontWeight.Medium)
            CvTextField(value = skillName, label = "Compétence", onChange = { skillName = it })

            Text("Niveau :", fontSize = 13.sp)
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                SkillLevel.entries.forEach { level ->
                    FilterChip(
                        selected = skillLevel == level,
                        onClick = { skillLevel = level },
                        label = { Text(level.label, fontSize = 11.sp) },
                        colors = FilterChipDefaults.filterChipColors(
                            selectedContainerColor = Gold,
                            selectedLabelColor = MaterialTheme.colorScheme.onPrimary
                        )
                    )
                }
            }

            Button(
                onClick = {
                    if (skillName.isNotBlank()) {
                        viewModel.addSkill(Skill(name = skillName, level = skillLevel))
                        skillName = ""
                    }
                },
                colors = ButtonDefaults.buttonColors(containerColor = Gold),
                modifier = Modifier.fillMaxWidth()
            ) {
                Icon(Icons.Default.Add, contentDescription = null, tint = MaterialTheme.colorScheme.onPrimary)
                Spacer(Modifier.width(8.dp))
                Text("Ajouter", color = MaterialTheme.colorScheme.onPrimary)
            }
        }

        HorizontalDivider(color = MediumGray)

        // Languages section
        Text("Langues", fontWeight = FontWeight.SemiBold, fontSize = 16.sp)

        cvData.languages.forEach { lang ->
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text("${lang.name} — ${lang.level}", fontSize = 14.sp)
                IconButton(onClick = { viewModel.removeLanguage(lang.id) }) {
                    Icon(Icons.Default.Close, contentDescription = null, tint = MaterialTheme.colorScheme.error, modifier = Modifier.size(16.dp))
                }
            }
        }

        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(MediumGray, RoundedCornerShape(12.dp))
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Text("Ajouter une langue", fontSize = 14.sp, fontWeight = FontWeight.Medium)
            CvTextField(value = langName, label = "Langue (ex: Français)", onChange = { langName = it })
            CvTextField(value = langLevel, label = "Niveau (ex: Natif, Courant, B2)", onChange = { langLevel = it })

            Button(
                onClick = {
                    if (langName.isNotBlank() && langLevel.isNotBlank()) {
                        viewModel.addLanguage(Language(name = langName, level = langLevel))
                        langName = ""; langLevel = ""
                    }
                },
                colors = ButtonDefaults.buttonColors(containerColor = Gold),
                modifier = Modifier.fillMaxWidth()
            ) {
                Text("Ajouter", color = MaterialTheme.colorScheme.onPrimary)
            }
        }

        Spacer(Modifier.height(80.dp))
    }
}

File: flutter_app/lib/screens/steps/skills_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../../models/cv_data.dart';
import '../../state/cv_model.dart';
import '../../theme/colors.dart';

/// Port de `ui/screens/SkillsScreen.kt`.
class SkillsScreen extends StatefulWidget {
  const SkillsScreen({super.key});

  @override
  State<SkillsScreen> createState() => _SkillsScreenState();
}

class _SkillsScreenState extends State<SkillsScreen> {
  final _skillNameCtrl = TextEditingController();
  final _langNameCtrl = TextEditingController();
  final _langLevelCtrl = TextEditingController();
  SkillLevel _skillLevel = SkillLevel.intermediate;

  @override
  void dispose() {
    _skillNameCtrl.dispose();
    _langNameCtrl.dispose();
    _langLevelCtrl.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final model = context.watch<CvModel>();
    final scheme = Theme.of(context).colorScheme;
    final data = model.cvData;

    return SingleChildScrollView(
      padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          const Text(
            'Compétences & Langues',
            style: TextStyle(
              fontSize: 20,
              fontWeight: FontWeight.bold,
              color: AppColors.gold,
            ),
          ),
          const SizedBox(height: 20),

          const Text(
            'Compétences',
            style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16),
          ),
          const SizedBox(height: 20),

          for (final skill in data.skills)
            Row(
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.stretch,
                    children: [
                      Padding(
                        padding: const EdgeInsets.only(right: 8),
                        child: Row(
                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
                          children: [
                            Text(
                              skill.name,
                              style: const TextStyle(fontSize: 14),
                            ),
                            Text(
                              skill.level.label,
                              style: const TextStyle(
                                fontSize: 12,
                                color: AppColors.gold,
                              ),
                            ),
                          ],
                        ),
                      ),
                      const SizedBox(height: 4),
                      LinearProgressIndicator(
                        value: skill.level.value,
                        minHeight: 4,
                        color: AppColors.gold,
                        backgroundColor: AppColors.mediumGray,
                      ),
                    ],
                  ),
                ),
                IconButton(
                  onPressed: () => model.removeSkill(skill.id),
                  icon: Icon(Icons.close, size: 16, color: scheme.error),
                ),
              ],
            ),

          const SizedBox(height: 20),

          // Formulaire d'ajout de compétence
          Container(
            decoration: BoxDecoration(
              color: AppColors.mediumGray,
              borderRadius: BorderRadius.circular(12),
            ),
            padding: const EdgeInsets.all(16),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              children: [
                const Text(
                  'Ajouter une compétence',
                  style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500),
                ),
                const SizedBox(height: 12),
                TextField(
                  controller: _skillNameCtrl,
                  decoration: const InputDecoration(
                    labelText: 'Compétence',
                    filled: false,
                    border: OutlineInputBorder(),
                  ),
                ),
                const SizedBox(height: 12),
                const Text('Niveau :', style: TextStyle(fontSize: 13)),
                const SizedBox(height: 8),
                Wrap(
                  spacing: 8,
                  children: [
                    for (final level in SkillLevel.values)
                      FilterChip(
                        selected: _skillLevel == level,
                        onSelected: (_) =>
                            setState(() => _skillLevel = level),
                        label: Text(
                          level.label,
                          style: const TextStyle(fontSize: 11),
                        ),
                        selectedColor: AppColors.gold,
                        labelStyle: TextStyle(
                          color: _skillLevel == level
                              ? scheme.onPrimary
                              : scheme.onSurface,
                        ),
                      ),
                  ],
                ),
                const SizedBox(height: 12),
                FilledButton.icon(
                  style: FilledButton.styleFrom(
                    backgroundColor: AppColors.gold,
                    foregroundColor: scheme.onPrimary,
                  ),
                  onPressed: () {
                    if (_skillNameCtrl.text.trim().isNotEmpty) {
                      model.addSkill(
                        Skill(
                          name: _skillNameCtrl.text.trim(),
                          level: _skillLevel,
                        ),
                      );
                      _skillNameCtrl.clear();
                    }
                  },
                  icon: const Icon(Icons.add),
                  label: const Text('Ajouter'),
                ),
              ],
            ),
          ),

          const SizedBox(height: 20),
          const Divider(color: AppColors.mediumGray),
          const SizedBox(height: 20),

          const Text(
            'Langues',
            style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16),
          ),
          const SizedBox(height: 20),

          for (final lang in data.languages)
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Text(
                  '${lang.name} — ${lang.level}',
                  style: const TextStyle(fontSize: 14),
                ),
                IconButton(
                  onPressed: () => model.removeLanguage(lang.id),
                  icon: Icon(Icons.close, size: 16, color: scheme.error),
                ),
              ],
            ),

          const SizedBox(height: 20),

          Container(
            decoration: BoxDecoration(
              color: AppColors.mediumGray,
              borderRadius: BorderRadius.circular(12),
            ),
            padding: const EdgeInsets.all(16),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.stretch,
              children: [
                const Text(
                  'Ajouter une langue',
                  style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500),
                ),
                const SizedBox(height: 12),
                TextField(
                  controller: _langNameCtrl,
                  decoration: const InputDecoration(
                    labelText: 'Langue (ex: Français)',
                    filled: false,
                    border: OutlineInputBorder(),
                  ),
                ),
                const SizedBox(height: 12),
                TextField(
                  controller: _langLevelCtrl,
                  decoration: const InputDecoration(
                    labelText: 'Niveau (ex: Natif, Courant, B2)',
                    filled: false,
                    border: OutlineInputBorder(),
                  ),
                ),
                const SizedBox(height: 12),
                FilledButton(
                  style: FilledButton.styleFrom(
                    backgroundColor: AppColors.gold,
                    foregroundColor: scheme.onPrimary,
                  ),
                  onPressed: () {
                    final name = _langNameCtrl.text.trim();
                    final level = _langLevelCtrl.text.trim();
                    if (name.isNotEmpty && level.isNotEmpty) {
                      model.addLanguage(Language(name: name, level: level));
                      _langNameCtrl.clear();
                      _langLevelCtrl.clear();
                    }
                  },
                  child: const Text('Ajouter'),
                ),
              ],
            ),
          ),

          const SizedBox(height: 80),
        ],
      ),
    );
  }
}
